sos/plugins/__init__.py

```diff
@@ -1123,6 +1123,21 @@ def check_process_by_name(self, process):
             return False
         return status
 
+    def get_process_pid(self, process):
+        """Returns a process PID if it exists.
+        If th process doesn't exist, returns -1"""
+        cmd_line_glob = "/proc/[0-9]*/cmdline"
+        cmd_line_paths = glob.glob(cmd_line_glob)
+        for path in cmd_line_paths:
```

Comment:
It's worth trying this out on a system with a very large number of threads running. Sometimes we've found external tools are quicker, but it's not a definite call (I had a feeling we had discussed this before, but I can't find that conversation now or what the conclusion was).

Anyway, it's not a big deal - many tools show a slowdown on those sorts of heavily loaded systems anyway.

At some point it may be useful to allow exact/inexact matching (i.e. an exact command line vs. any instance of a program), but it's not needed today so there's no need to implement it.

Reply:
OK, leaving this as is.

Reply:
This finds only the first process, but we need all of them.